### DISCUSSION TITLE
Similar decay rates and peaks for all channels

### DISCUSSION THREAD

**Discussion by braggbear** [sibylhe/mmm_stan#15] [Q&A]
Hello. When I run your model (with your data) on my machine and check the summary, I see that each channel has a mean decay rate of 0.5 and a mean peak around 1.5. In reality, shouldn't the various channels have different decay and peaks? Are they similar because the model can't determine from the data and is just using the priors?  If so, any thoughts on how we could handle this? Thanks.

**Comment by sibylhe** [comment]
Yes, further work needs to be done with adstock. I've thought of two approaches.

**Approach 1: Use more specific priors for each channel**
Current priors are decay ~ beta(3,3); peak ~ uniform(0, 4), length=8 for all channels.
We may set specific prior for each channel. e.g., sem is expected to have a low decay, immediate effect, set its decay ~ half normal(0.1, 0.1); peak ~ uniform(0, 1), length=2.

**Approach 2: Grid search predetermined adstock parameters**
param_grid = {
 sem: {'D': range(0.1, 0.5, 0.1), 'P': [0, 1], 'L': [2]},
 dm: {'D': range(0.1, 0.8, 0.1), 'P': [0, 1, 2, 4], 'L': [8]},
 ...
}
Apply different sets of adstock parameters to media variables, train models with adstocked media variables, select the most accurate and reasonable model.
